src/content/post/fastrace-a-modern-approach-to-distributed-tracing-in-rust.mdx

```diff
@@ -0,0 +1,374 @@
+
+`fastrace` takes a different approach to tracing in Rust. Its core design principles include:
+
+### 1. Zero-cost Abstraction
+
+`fastrace` is designed as a true zero-cost abstraction. When disabled, instrumentation compiles away completely, resulting in no runtime overhead. This makes it ideal for libraries concerned about performance.
+
+### 2. Ecosystem Compatibility
+
+`fastrace` focuses exclusively on distributed tracing. Through its composable design, it integrates seamlessly with the existing Rust ecosystem, including compatibility with the standard `log` crate. This architectural approach allows libraries to implement comprehensive tracing while preserving their users' freedom to use their preferred logging setup.
+
+### 3. Simplicity First
+
+The API is designed to be intuitive and require minimal boilerplate, focusing on the most common use cases while still providing extensibility when needed.
+
+### 4. Insanely Fast
+
+![Fastrace Performance](assets/trace-100-spans.png)
+
+`fastrace` is designed for high-performance applications. It can handle massive amount of spans per second with minimal impact on CPU and memory usage.
+
+### 5. Ergonomics for both Applications and Libraries
+
+Libraries can use `fastrace` without imposing performance penalties when not needed:
+
+```rust
+// In a library crate:
+use fastrace::prelude::*;
+
+#[fastrace::trace]  // Zero-cost when the application doesn't enable the 'enable' feature
+pub fn process_data(data: &[u8]) -> Result<Vec<u8>, Error> {
+    // Library uses standard log crate
+    log::debug!("Processing {} bytes of data", data.len());
+    
+    // ...more code...
+}
+```
+
+The key point here is that libraries should include fastrace without enabling any features:
+
+```toml
+[dependencies]
+fastrace = "0.7"  # No 'enable' feature
+```
+
+When an application uses this library and doesn't enable the 'enable' feature of fastrace:
+- All tracing code is completely optimized away at compile time
+- Zero runtime overhead is added to the library
+- No impact on performance-critical code paths
+
+When the application does enable tracing via the 'enable' feature:
+- Instrumentation in the dedicated library becomes active
+- Spans are collected and reported
+- The application gets full visibility into library behavior
+
+This is a significant advantage over other tracing solutions that either always impose overhead or require libraries to implement complex feature-flag systems.
+
+### 6. Seamless Context Propagation
+
+`fastrace` provides companion crates for popular frameworks that handle context propagation automatically:
+
+```rust
+// For HTTP clients with reqwest
+let response = client.get(&format!("https://user-service/users/{}", user_id))
+    .headers(fastrace_reqwest::traceparent_headers())
+    .send()
+    .await?;
+
+// For gRPC servers with tonic
+Server::builder()
+    .layer(FastraceServerLayer)  // Automatically extracts context from incoming requests
```

Comment:
```suggestion
    .layer(fastrace_tonic::FastraceServerLayer)  // Automatically extracts context from incoming requests
```

I guess it's from`fastrace_tonic`? If so, let's make it clear. Ditto the following bridge.